File: interpreter/interpreter/token.go
// pascal/token.go
package pascal

type TokenType int

const (
	INTEGER TokenType = iota
	PLUS
	MINUS
	MUL
	DIV
	LPAREN
	RPAREN
	ID
	ASSIGN
	BEGIN
	END
	SEMI
	DOT
	EOF
)

type Token struct {
	Type  TokenType
	Value string
}

func NewToken(tokenType TokenType, value string) Token {
	return Token{
		Type:  tokenType,
		Value: value,
	}
}

func (t Token) String() string {
	return "Token(" + t.Value + ")"
}

File: interpreter_full/token.py
from enum import Enum, auto

class TokenType(Enum):
    NUMBER = auto()
    OPERTATOR = auto()
    LPAREN = auto()
    RPAREN = auto()
    EOL = auto()

class Token():

    def __init__(self, type_: TokenType, value: str):
        self.type_ = type_
        self.value = value

    def __str__(self):
        return f'{self.__class__.__name__}){self.type_}, {self.value}'